provider/pkg/provider/provider.go

```diff
@@ -521,6 +528,9 @@ func (p *cfnProvider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*
 	// Filter null properties from the inputs.
 	newInputs = filterNullProperties(newInputs)
 
+	// Auto-name fields if not already specified
+	setDefaultName(urn, spec, olds, newInputs)
```

Comment:
Won't this overwrite an explicit name if it's set in user's code?

Reply:
As written, the following will make sure to not overwrite names if specified.
```
if _, ok := olds[resource.PropertyKey(sdkName)]; ok {
        continue
}
```

Reply:
yup fixed now and added a test.